### PR TITLE
Fix panic due to range out of bounds in txt record parsing

### PR DESCRIPTION
**Pull request by Raphiiko**
I noticed the following panic in my crash reporting earlier:

```
range end index 303 out of range for slice of length 288 (C:\...\src\service_info.rs:645:25)
```

I'm not entirely sure if this would be the correct solution, but I've made an attempt.

### CODE REVIEW

File: src/service_info.rs [hunk 1]
```diff
@@ -642,7 +642,11 @@ fn decode_txt(txt: &[u8]) -> Vec<TxtProperty> {
         }
         offset += 1; // move over the length byte
 
-        let kv_bytes = &txt[offset..offset + length];
+        let offset_end = offset + length;
+        if offset_end > txt.len() {
+            break; // Would be out of range, skipping this property.
```
Comment (keepsimple1) [inline src/service_info.rs:647]: Make sense! Could you also add a log of `offset_end` and `txt.len()` when this happens?  I would say `error!()` level.  And if `error!()` doesn't work for you, `debug!()` level is also fine. 

Can you also add a unit test in this file inside `mod tests` for this negative case?  A simple test input would be fine.

**Review by keepsimple1 — COMMENTED** [review]
Thanks for your PR! Fix looks nice, only a minor comment inline.

File: src/service_info.rs (hunk 1 above)
Comment (Raphiiko) [reply src/service_info.rs]: Gotcha, I've give given it a go!

**Review by keepsimple1 — APPROVED** [review]
Looks good! Thanks!
